frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -912,42 +930,216 @@ function ReaderFooter:addToMainMenu(menu_items)
+                        end
+                    },
+                },
+            },
+            {
+                text_func = function()
+                    local text = _("static margins (10)")
+                    if self.settings.progress_margin_width == Screen:scaleBySize(0) then
+                        text = _("no margins (0)")
+                    end
+                    if self.settings.progress_margin and not self.ui.document.info.has_pages then
+                        local margins = self.ui.document:getPageMargins()
+                        text = T(_("same as book margins (%1)"), math.floor((margins.left + margins.right)/2))
+                    end
+                    return T(_("Margins: %1"), text)
+                end,
+                enabled_func = function()
+                    return not self.settings.disable_progress_bar and self.settings.progress_bar_position ~= nil
                 end,
+                separator = true,
+                sub_item_table = {
+                    {
+                        text = _("No margins (0)"),
+                        checked_func = function()
+                            return self.settings.progress_margin_width == Screen:scaleBySize(0)
+                                and not self.settings.progress_margin
+                        end,
+                        callback = function()
+                            self.settings.progress_margin_width = Screen:scaleBySize(0)
+                            self.settings.progress_margin = false
+                            self:refreshFooter(nil, true)
+                        end,
+                    },
+                    {
+                        text = _("Static margins (10)"),
+                        checked_func = function()
+                            return self.settings.progress_margin_width == Screen:scaleBySize(10)
+                                and not self.settings.progress_margin
+                                -- if same as book margins is selected in document with pages (pdf) we enforce static margins
+                                or (self.ui.document.info.has_pages and self.settings.progress_margin)
+                        end,
+                        callback = function()
+                            self.settings.progress_margin_width = Screen:scaleBySize(10)
+                            self.settings.progress_margin = false
+                            self:refreshFooter(nil, true)
+                        end,
+                    },
+                    {
+                        text_func = function()
+                            if self.ui.document.info.has_pages then
+                                return _("Same as book margins")
+                            end
+                            local margins = self.ui.document:getPageMargins()
+                            return T(_("Same as book margins (%1)"), math.floor((margins.left + margins.right)/2))
```

Comment:
Maybe better show in the menu scaled value

```lua
                        text_func = function()
                            return T(_("Static margins (%1)"), Screen:scaleBySize(10))
                        end,
```

Reply:
Whatever you end up with, if it's a scaled value, that fact should be made explicit (with a px suffix), like we did in the feedback popup when setting the CRe margins ;).

Reply:
Better, I don't think so (we display font size unscaled, and other things too), but simpler possibly :)
Or may be just don't fetch them from crengine.
The only place that seems to feed them to crengine is:
https://github.com/koreader/koreader/blob/b75fec4971c1b2fa46e9a69cf72c2275fc10d35f/frontend/apps/reader/modules/readertypeset.lua#L502-L512
And this function always get them unscaled.

It's a bit unclear if this event is always sent (it's sometimes called directly, so the duplication mentionned below - and this function `return true`, possibly  stopping propagation).
But we have ReaderDogear (that is loaded by ReaderView, just like ReaderFooter is), that may get the event before ReaderTypeset, that already does react to margins - so, you may do as ReaderDogear does in ReaderFooter:
https://github.com/koreader/koreader/blob/b75fec4971c1b2fa46e9a69cf72c2275fc10d35f/frontend/apps/reader/modules/readerdogear.lua#L43-L58
https://github.com/koreader/koreader/blob/b75fec4971c1b2fa46e9a69cf72c2275fc10d35f/frontend/apps/reader/modules/readerdogear.lua#L60-L73
(That @NiLuJe code will then be triplicated - but I won't mention it as it may get complicated to do differently :)

Reply:
So, the idea is that instead of fetching them each time, you would store them in `self`.
Read their initial value in onReadSettings, store them.
Get the new value when it's updated, by reacting to `onSetPageMargins` (you already do it for `onSetPageHorizMargins`, but `ReaderTypeset:onSetPageHorizMargins` will send `self.ui:handleEvent(Event:new("SetPageMargins", self.unscaled_margins)` - so no need for your `onSetPageHorizMargins `then).